ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -183,6 +275,61 @@ def _create_sessions_core(
                 )
             yield typed_session_infos
 
+    def _get_connection_core(
+        self,
+        session_type: Type[TSession],
+        pin_or_relay_names: Union[str, Iterable[str], None] = None,
+        sites: Union[int, Iterable[int], None] = None,
+        instrument_type_id: Optional[str] = None,
+    ) -> TypedConnection[TSession]:
+        results = self._get_connections_core(
+            session_type, pin_or_relay_names, sites, instrument_type_id
+        )
+        if len(results) > 1:
+            raise ValueError(
+                "Too many reserved connections matched the specified criteria. "
+                f"Expected single connection, got {len(results)} connections."
+            )
+        return results[0]
+
+    def _get_connections_core(
+        self,
+        session_type: Type[TSession],
+        pin_or_relay_names: Union[str, Iterable[str], None] = None,
+        sites: Union[int, Iterable[int], None] = None,
+        instrument_type_id: Optional[str] = None,
+    ) -> Sequence[TypedConnection[TSession]]:
+        pin_order = _to_iterable(pin_or_relay_names, self._reserved_pin_or_relay_names)
+
+        if sites == SITE_ALL_SITES:
+            site_order = self._reserved_sites
+        else:
+            site_order = _to_iterable(sites, self._reserved_sites)
+            if SITE_SYSTEM_PINS in self._reserved_sites and SITE_SYSTEM_PINS not in site_order:
+                site_order = site_order + [SITE_SYSTEM_PINS]
+
+        instrument_type_id_order = _to_iterable(
+            instrument_type_id, self._reserved_instrument_type_ids
+        )
+
+        results: List[TypedConnection[TSession]] = []
+        for instrument_type in instrument_type_id_order:
```

Comment:
Instrument type should not be the top-level sort because that overrides the specified site/pin order in unexpected ways.

@pbirkhol-ni I realize the LV version requires the instrument type id, so it doesn't even have this situation.

Reply:
I commented on this below, but yes, currently we don't have a generic Get Connections for LabVIEW. When we do, I haven't decided whether instrument type id will be required or optional.

Reply:
Instrument type id is now the last-level, "tiebreaker" sort.